web-marketplace/src/pages/ProfileEdit/ProfileEdit.Settings.tsx

```diff
@@ -0,0 +1,98 @@
+import { useState } from 'react';
+
+import ErrorBanner from 'web-components/lib/components/banner/ErrorBanner';
+
+import { useAuth } from 'lib/auth/auth';
+import { apiServerUrl } from 'lib/env';
+import { useSignArbitrary } from 'lib/wallet/hooks/useSignArbitrary';
+import { useWallet } from 'lib/wallet/wallet';
+
+import { AccountConnectWalletModal } from 'components/organisms/AccountConnectWalletModal/AccountConnectWalletModal';
+import { useLoginData } from 'components/organisms/LoginButton/hooks/useLoginData';
+import { UserAccountSettings } from 'components/organisms/UserAccountSettings/UserAccountSettings';
+import {
+  SocialProviderInfo,
+  WalletProviderInfo,
+} from 'components/organisms/UserAccountSettings/UserAccountSettings.types';
+
+import { useConnectKeplrWallet } from './hooks/useConnectKeplrWallet';
+
+export const ProfileEditSettings = () => {
+  const [error, setError] = useState<unknown>(undefined);
+  const { authenticatedAccounts, activeAccount } = useAuth();
+  const { wallet } = useWallet();
+
+  const {
+    connecting,
+    isModalOpen,
+    modalState,
+    onButtonClick,
+    onModalClose,
+    qrCodeUri,
+    walletsUiConfig,
+  } = useLoginData();
+
+  const signArbitrary = useSignArbitrary({
+    setError,
+  });
+
+  // Google accounts
+  const googleAccounts = authenticatedAccounts?.filter(
+    account => !!account?.email,
+  );
+
+  const googleSocialProviders: SocialProviderInfo[] =
+    googleAccounts?.map(account => ({
+      providerName: `Google`,
+      address: `${account?.email}`,
+      disconnect: () => undefined,
+    })) ?? [];
+  const connectGoogleAccount = () =>
+    (window.location.href = `${apiServerUrl}/marketplace/v1/auth/google`);
+
+  // Keplr accounts
+  const hasKeplrAccount = !!activeAccount?.addr;
+  const isCurrrentAddressAuthenticated =
+    authenticatedAccounts?.some(account => account?.addr === wallet?.address) ??
+    false;
+  const walletProviderInfo: WalletProviderInfo = hasKeplrAccount
+    ? { address: String(activeAccount?.addr), disconnect: () => void 0 }
+    : { connect: onButtonClick };
+
+  useConnectKeplrWallet({
+    setError,
```

Comment:
only /wallet-connect should be called when clicking the "connect" button
currently when I click it, I get the Prove Ownership Keplr popup multiple times and /login is called, so the current account is changed to the one corresponding to the wallet address

Reply:
I just double-checked that and we need an account to request a nonce, hence the `/login` call, otherwise we get `{"error":"Account not found for the given address"}` from the nonce call.

Reply:
Regarding the multiple popup, there's one for the login and one to add the wallet to the account. If there's a way to get a nonce without having an address associated with an account let me know.

Reply:
> I just double-checked that and we need an account to request a nonce, hence the `/login` call, otherwise we get `{"error":"Account not found for the given address"}` from the nonce call.

you can get the nonce from the graphql account query directly based on the account id and have it available in activeAccounts from useAuth

Reply:
see https://github.com/regen-network/regen-web/pull/2206/commits/f8f0001f636685788987fe8f71010ea8d0c77d2a